From 4ca97fb094f207989c63a60a5c09e1aa8023ceac Mon Sep 17 00:00:00 2001
From: Micha? Janiszewski <michal1x.janiszewski@intel.com>
Date: Wed, 7 Mar 2018 20:10:06 +0800
Subject: [PATCH 1167/2367] x86/paravirt: don't try getting cpu_khz if
 unavailable

Compliments commit 7f95f91ade68:
x86:cwp: add cpu_khz pv ops to get cpu khz from pv method

Change-Id: I983481f9b3fa2dac5bbfaaa6b037727bb6ea5aac
Signed-off-by: Micha? Janiszewski <michal1x.janiszewski@intel.com>
Reviewed-on: https://android.intel.com/621406
---
 arch/x86/include/asm/paravirt.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/x86/include/asm/paravirt.h b/arch/x86/include/asm/paravirt.h
index f9be9e7..046e2c1 100644
--- a/arch/x86/include/asm/paravirt.h
+++ b/arch/x86/include/asm/paravirt.h
@@ -275,6 +275,8 @@ static inline void slow_down_io(void)
 
 static inline unsigned long cpu_khz_from_paravirt(void)
 {
+	if (pv_cpu_ops.cpu_khz == NULL)
+		return 0;
 	return PVOP_CALL0(unsigned long, pv_cpu_ops.cpu_khz);
 }
 
-- 
2.7.4

